### DISCUSSION TITLE
How to overide bot appsetting in web app setting

### DISCUSSION THREAD

**Discussion by TonyZhao414** [microsoft/BotFramework-Composer#9702] [Q&A]
Hi, I'm trying to write logs of my bot to app insights, which is deployed to a web app. But unforunately there's only activity log (like connection start, listening to host) in application insights linked to my web app, and I could see nothing in customEvent table. After adding this connectionString setting below in my local appsetting.json file, I was able to commit logs to app insights with my question and answer in local bot. So I'm wondering how could I overide this setting in my web app or anywhere else. Would you please help answer or point out if there's something wrong with my current deployment causing only activity log. I was adding APPLICATIONINSIGHTS_CONNECTION_STRING and APPINSIGHTS_INSTRUMENTATIONKEY in web app setting, but that doesn't seem to work. Also, I tried runtimeSettings__telemetry__options__connectionString. Thanks.

"runtimeSettings": {
    "telemetry": {
      "logActivities": true,
      "logPersonalInformation": false,
      "options": {
        "connectionString": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
      }
    }
  },

**Comment by OEvgeny** [comment]
To my knowledge it should be not `runtimeSettings__telemetry__options__connectionString` but `telemetry:options:connectionString`:

Try the following:
1. Go to your App Service in the Azure portal and navigate to the "Configuration" tab.
2. Click on the "New application setting" button to add a new configuration setting.
3. Set the "Name" field to telemetry:options:connectionString.
4. Set the "Value" field to the connection string for your Application Insights instance.
5. Click on the "OK" button to save the configuration setting.

**Reply by TonyZhao414** [reply]
Hi @OEvgeny, thanks for your help.

I tried to use 'telemetry:options:connectionString' in web app setting, then I could see logs in customEvent table in application insights. However, there are only logs for 'Activity', like chatbot status. While I'm using local emulator, there are 'GeneratorResult' including bot message. Do you know what I'm missing, or is there any other right log table I should look into? Thanks.

'Activity' logs
<img width="277" alt="image" src="https://github.com/microsoft/BotFramework-Composer/assets/10001849/99c1966f-461d-418d-8427-dd65ccded20c">

'GeneratorResult' logs
<img width="477" alt="image" src="https://github.com/microsoft/BotFramework-Composer/assets/10001849/382469ec-3617-40ce-9389-a37d714fb5c4">

**Reply by OEvgeny** [reply]
To me this should be enough to have these events. This also might be related: https://github.com/microsoft/BotFramework-Composer/issues/9303#issuecomment-1185210117
